examples/Training/java/src/training/HowToUseTables.java

```diff
@@ -114,6 +123,25 @@ private void createTable()
             table.addData(newRow);
             OriginalFile file = table.getOriginalFile(); // if you need to interact with the table
             file = new OriginalFileI(file.getId(), false);
+
+            DataManagerFacility dm = gateway.getFacility(DataManagerFacility.class);
+
+            FileAnnotation annotation = new FileAnnotationI();
+            annotation.setFile(file);
+            annotation.setNs(omero.rtypes
+                    .rstring(omero.constants.namespaces.NSBULKANNOTATIONS.value));
+            
+            //The following saveAndReturnObject call can be removed in bulk annotation scenarios
+            //The server call is made when a link is created and that handles both the annotation and the link together
+            annotation = (FileAnnotation) dm.saveAndReturnObject(ctx, annotation);
+
+            //now link the image and the annotation
+            ImageAnnotationLink link = new ImageAnnotationLinkI();
+            link.setChild(annotation);
+            link.setParent(image.asImage());
```

Comment:
This will thrown a NPE, because `image` hasn't been loaded. You could for example load it in the constructor, before calling `createTableandLinkToImage()` after line 193.
```
// load the image
BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
image = browse.getImage(ctx, imageId);
```